### PR TITLE
Barrier Option Price XLA compatibility.

### PR DESCRIPTION
**Pull request by DevarakondaV**
Hey Guys,

Added XLA compatibility for the "barrier_price" function. Also changed the condition statement to run function. Let me know if any additional changes are required. Sorry about the delay on this! Feel free to pull me in for any other features as I have more free time now!

### CODE REVIEW

File: tf_quant_finance/black_scholes/vanilla_prices.py [hunk 1]
```diff
@@ -293,9 +293,9 @@ def barrier_price(*,
     the barriers option under black scholes.
 
   """
-  if (continuous_dividends is None) == (cost_of_carries is None):
-    raise ValueError('At most one of continuous_dividends and cost of carries '
-                     'may be supplied')
+  if (continuous_dividends is not None) == (cost_of_carries is not None):
```
Comment (cyrilchim) [inline tf_quant_finance/black_scholes/vanilla_prices.py:296]: actually continuous dividends and cost_of_carries can both be None and should be treated like [here](https://github.com/google/tf-quant-finance/blob/848762e3da280601acdb571c2b6e971aa6670418/tf_quant_finance/black_scholes/vanilla_prices.py#L150)

File: tf_quant_finance/black_scholes/vanilla_prices.py [hunk 2]
```diff
@@ -397,7 +398,7 @@ def barrier_price(*,
     strikes_term = call_or_put * strikes * discount_rates_exponent
 
     # rank is used to stack elements and reduce_sum
-    strike_rank = tf.rank(strikes)
+    strike_rank = strikes.get_shape().ndims
```
Comment (cyrilchim) [inline tf_quant_finance/black_scholes/vanilla_prices.py:401]: Instead:
strikes.shape.rank

**Comment by cyrilchim** [comment]
Hi Vishnu,

Could you please let me know if you need any help resolving the comments?

Best,
Cyril

**Comment by DevarakondaV** [comment]
@cyrilchim Woops, I missed your first update. Sorry about that. Just pushed the changed!

**Pull request merged**